signac/contrib/project.py

```diff
@@ -506,6 +510,8 @@ def detect_schema(self, exclude_const=False, subset=None, index=None):
         statepoint_index = self.build_job_statepoint_index(exclude_const=exclude_const, index=index)
         return ProjectSchema.detect(statepoint_index)
 
+    @deprecated(deprecated_in="1.3", removed_in="2.0", current_version=__version__,
+                details="Use find_jobs().ids instead.")
     def find_job_ids(self, filter=None, doc_filter=None, index=None):
         """Find the job_ids of all jobs matching the filters.
```

Comment:
If we're deprecating both `build_job_search_index` and `find_job_ids`, can we get rid of the `JobSearchIndex` class as well? Users who really want to use indices should probably just use Collections anyway.

Reply:
Yes, I think that would be a good idea, although let's check with others before we do something this drastic.

Reply:
This still needs to be done. @tommy-waltmann The change you want looks like this, roughly (comments are only for your guidance, don't include them):
```python
# Move the implementation to this private method
def _find_jobs_ids(...):
    # do stuff - copy the implementation from the current public method
    pass

# Keep a public method that is deprecated
@deprecated(...)
def find_job_ids(...):
    return _find_job_ids(...)
```